### PR TITLE
gnomeExtensions: fix ddterm

### LINKED ISSUE #125363: gnome extension "ddterm" does not work
**Describe the bug**
The "ddterm" extension does not work, because it lacks some NixOS-specific packaging magic having to do with `gjs`, typelibs etc (I don't understand anything about it, just keyword-googled #40474)

**To Reproduce**
1. add `gnomeExtensions.ddterm` to `environment.systemPackages`
2. login into Gnome
3. press F12
4. nothing visible happens.  "/run/current-system/sw/share/gnome-shell/extensions/ddterm@amezin.github.com/com.github.amezin.ddterm: line 5: exec: gjs: not found" appears in the user journal

**Notify maintainers**
@piegamesde  

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.37, NixOS, 21.05.596.3a2e0c36e79 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(cmm): `"home-manager-21.05, nixpkgs-21.05.596.3a2e0c36e79, nixpkgs-unstable-21.11pre292293.84aa23742f6"`
 - channels(root): `"nixos-21.05.596.3a2e0c36e79, nixos-unstable-21.11pre292748.6933d068c5d"`
 - nixpkgs: `/home/cmm/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  gnomeExtensions
  gnome40Extensions
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by cmm**
It needs gjs, zenity & vte.

Fixes #125363.

###### Motivation for this change
The Gnome extension "ddterm" is part of Nixpkgs but does not work.

###### Things done

- Built on platform(s)
  - [v] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/gnome/extensions/extensionOverrides.nix [hunk 1]
```diff
@@ -57,4 +61,21 @@ super: lib.trivial.pipe super [
         --replace "GLib.build_filenamev([GLib.DIR_SEPARATOR_S, 'usr', 'share', 'touchegg', 'touchegg.conf'])" "'${touchegg}/share/touchegg/touchegg.conf'"
     '';
   }))
+
+  (patchExtension "ddterm@amezin.github.com" (old: {
+    # Requires gjs, zenity & vte via the typelib
+    nativeBuildInputs = old.nativeBuildInputs or [] ++ [
+      wrapGAppsHook
+      gobject-introspection vte
```
Comment (jtojnar) [inline pkgs/desktops/gnome/extensions/extensionOverrides.nix:69]: `vte` should go to `buildInputs` since it is a host dependency.

**Review by piegamesde — CHANGES_REQUESTED** [review]
Tested and works in my VM. One thing left though: please ensure that the entries are sorted alphabetically.

**Comment by piegamesde** [comment]
Um, did anything change with the latest push?

**Comment by cmm** [comment]
> Um, did anything change with the latest push?

I was unclear on which "entries" you were referring to.  have I gotten it right this time?

**Pull request merged**